Include headless feature in 1.7 release

jrwdunham opened the pull request:
Fixes archivematica.git subpart of #878.

sevein committed:
Dashboard: add indexless setting

sevein committed:
Use indexless mode by default

sevein committed:
MCPClient: honor DISABLE_SEARCH_INDEXING

jrwdunham committed:
Dashboard: reflect DISABLE_SEARCH_INDEXING in GUI

If Archivematica has been deployed with Elasticsearch indexing disabled,
then the Backlog, Appraisal, and Archival storage tabs will now no
longer be displayed. In addition, the SIP Arrange pane in the Ingest tab
is no longer displayed. Finally, the status of ES indexing is also indicated
under Administration > General.

jrwdunham committed:
Dashboard: fix processing config for indexless

If ES indexing is turned off (i.e., setting `DISABLE_SEARCH_INDEXING` is set
to `True`), then the processing config edit page will no longer display
the "Send to backlog" option at the "Create SIP(s)" choice.

jrwdunham committed:
MCPServer: fix choices for indexless

If ES indexing is disabled, then the MCPServer will no longer cause the
"Send to backlong" option to be available at the "Create SIP(s)" choice
point.

jrwdunham committed:
Add common abilities with choice dependencies

New abilities.py module in archivematicaCommon defines abilities of an
Archivematica instance as a whole. An ability can have dependencies. For
example, the Backlog feature is dependent on the Search ability. If the Search
(i.e., Elasticsearch indexing) ability is disabled, then the Backlog tab should
be unavailable; in addition, the choice to send a transfer to backlog (either
via pre-configuration or via real-time choice in the GUI) should not be
available either.

Some minor formatting and i18n changes were also made.

jrwdunham committed:
Enhance appconfig to allow multiple equivalent options

The `appconfig.py::Config` class accepts an `attrs` dict argument whose values
can now contain lists of dicts as opposed to just single dicts as previsouly.
This allows one to specify multiple configuration options as equivalent and
reduce them to a single option. Using this functionality, the
`disable_search_indexing` and `search_enabled` config options are declared to
be equivalent, the one being the inverse of the other. The resulting settings
attribute is accessed as `settings.SEARCH_ENABLED`.

jrwdunham committed:
archivematicaCommon: add tests for appconfig

jrwdunham committed:
Remove SEARCH_ENABLED_DEFAULT constant from settings

sevein commented:
Can you add the new config to the READMEs?

On Jan 29, 2018 16:54, "Joel Dunham" <notifications@github.com> wrote:

> Fixes archivematica.git subpart of #878
> <https://github.com/artefactual/archivematica/issues/878>.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/artefactual/archivematica/pull/901
> Commit Summary
>
>    - Dashboard: add indexless setting
>    - Use indexless mode by default
>    - MCPClient: honor DISABLE_SEARCH_INDEXING
>    - Dashboard: reflect DISABLE_SEARCH_INDEXING in GUI
>    - Dashboard: fix processing config for indexless
>    - MCPServer: fix choices for indexless
>    - Add common abilities with choice dependencies
>    - Enhance appconfig to allow multiple equivalent options
>    - archivematicaCommon: add tests for appconfig
>    - Remove SEARCH_ENABLED_DEFAULT constant from settings
>
> File Changes
>
>    - *M* src/MCPClient/lib/clientScripts/elasticSearchIndexProcessTrans
>    fer.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-0>
>    (2)
>    - *M* src/MCPClient/lib/clientScripts/indexAIP.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-1>
>    (2)
>    - *M* src/MCPClient/lib/clientScripts/post_store_aip_hook.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-2>
>    (4)
>    - *M* src/MCPClient/lib/clientScripts/removeAIPFilesFromIndex.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-3>
>    (3)
>    - *M* src/MCPClient/lib/settings/common.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-4>
>    (9)
>    - *M* src/MCPServer/lib/linkTaskManagerChoice.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-5>
>    (10)
>    - *M* src/MCPServer/lib/settings/common.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-6>
>    (7)
>    - *A* src/archivematicaCommon/lib/abilities.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-7>
>    (36)
>    - *M* src/archivematicaCommon/lib/appconfig.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-8>
>    (44)
>    - *M* src/archivematicaCommon/lib/env_configparser.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-9>
>    (4)
>    - *A* src/archivematicaCommon/tests/test_appconfig.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-10>
>    (65)
>    - *M* src/dashboard/src/components/administration/forms.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-11>
>    (4)
>    - *M* src/dashboard/src/components/archival_storage/views.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-12>
>    (3)
>    - *M* src/dashboard/src/components/backlog/views.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-13>
>    (6)
>    - *A* src/dashboard/src/main/context_processors.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-14>
>    (5)
>    - *M* src/dashboard/src/settings/base.py
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-15>
>    (8)
>    - *M* src/dashboard/src/templates/administration/general.html
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-16>
>    (8)
>    - *M* src/dashboard/src/templates/appraisal/appraisal.html
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-17>
>    (13)
>    - *M* src/dashboard/src/templates/archival_storage/list.html
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-18>
>    (187)
>    - *M* src/dashboard/src/templates/backlog/backlog.html
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-19>
>    (11)
>    - *M* src/dashboard/src/templates/ingest/grid.html
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-20>
>    (42)
>    - *M* src/dashboard/src/templates/layout.html
>    <https://github.com/artefactual/archivematica/pull/901/files#diff-21>
>    (10)
>
> Patch Links:
>
>    - https://github.com/artefactual/archivematica/pull/901.patch
>    - https://github.com/artefactual/archivematica/pull/901.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/artefactual/archivematica/pull/901>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AAlA-zTSVMCDJ9N2IqQLRW5E_1MnYBL7ks5tPmhGgaJpZM4RxmSo>
> .
>

jrwdunham committed:
Update install READMEs with search_enabled setting

jrwdunham committed:
Fix search_enabled iboolean mix-up

Previously the SEARCH_ENABLED setting was being set to the inverse
boolean (`iboolean`) of the corresponding environment variable (e.g.,
ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_SEARCH_ENABLED) and to the same thing
as the inverse env var (e.g.,
ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_DISABLEELASTICSEARCHINDEXING). This
was backwards and has been fixed by this commit.

jrwdunham committed:
Fix search_enabled in templates

jrwdunham commented:
Cherry-picked to stable/1.7.x as b4b4e0d8bd1cd927803279c4dc9e5d9b0875fdfa through 3794deb8353d3cef847ba9d2cf8538665924ada6.
